### PR TITLE
ramips-mt7620: add support for Xiaomi MiWifi Mini

### PR DESCRIPTION
**Pull request by Dark4MD**
* [x]  must be flashable from vendor firmware
  
 * There are two ways to flash. Both are written in the OpenWRT Wiki. One via an Code Injection Bug and one via an Account on miwifi.com
* https://openwrt.org/toh/xiaomi/mini#quick_openwrt_installation

* [x]  must support upgrade mechanism

  * [x]  must have working sysupgrade
    
    * [x]  must keep/forget configuration (if applicable)
      _think `sysupgrade [-n]` or `firstboot`_
  * [x]  must have working autoupdate
    _usually means: gluon profile name must match image name_

* wired network
  
  * [x]  should support all network ports on the device
  * [x]  must have correct port assignment (WAN/LAN)

* wifi (if applicable)
  
  * [x]  association with AP must be possible on all radios
  * [x]  association with 802.11s mesh must be working on all radios
  * [x]  ap/mesh mode must work in parallel on all radios
* mesh is working but it depends on PR #1626 to get it working on both radios, without it only mesh on radio0 (the extra wifi chip) is working.

* led mapping
 
  * power/sys led
    
    * [x]  lit while the device is on
    * [x]  should display config mode blink sequence
      (https://gluon.readthedocs.io/en/latest/features/configmode.html)

  * radio leds
    
    * [ ]  should map to their respective radio
    * [ ]  should show activity

      * device has only one LED for power status

  * switchport leds
    
    * [ ]  should map to their respective port (or switch, if only one led present)

    * [ ]  should show link state and activity

      * device has only one LED for power status

* [x]  reset/wps button must return device into config mode

* it just have an reset button and it works fine

* [x]  primary mac should match address on device label (or packaging) (https://gluon.readthedocs.io/en/latest/dev/hardware.html#notes)

* fixed

### CODE REVIEW

**Comment by blocktrron** [comment]
#1626 should fix the issues with the 2.4 radio, can you confirm that?

**Comment by achterin** [comment]
@ambassador86 and I experimented with similar devices (TP-Link Archer C20v1 & ASUS RT-AC51U) and had the same issue. If you disable the 5GHz chip completely (edit dts-boardfile in OpenWrt) the 2.4GHz radio is capable of running AP & Mesh-Mode in parallel.
@blocktrron i will test this tonight on my asus device and report back.

**Comment by Dark4MD** [comment]
@blocktrron i will try this and report back as soon as i got it running with the potential fix and yes, the image name is the one i get as an output with "lua -e 'print(require("platform_info").get_image_name())'"

**Comment by Dark4MD** [comment]
After applying the Fix in #1626 mesh on both radios is working.

File: package/gluon-core/luasrc/lib/gluon/upgrade/010-primary-mac [hunk 1]
```diff
@@ -28,6 +28,8 @@ if platform.match('ar71xx', 'generic', {'tl-wdr3600', 'tl-wdr4300',
   table.insert(try_files, 1, '/sys/class/ieee80211/phy1/macaddress')
 elseif platform.match('ramips', 'mt7621', {'dir-860l-b1'}) then
   table.insert(try_files, 1, '/sys/class/ieee80211/phy1/macaddress')
+elseif platform.match('ramips', 'mt7620', {'miwifi-mini'}) then
+  table.insert(try_files, 1, '/sys/class/net/eth0/address')
```
Comment (rotanid) [inline package/gluon-core/luasrc/lib/gluon/upgrade/010-primary-mac]: are you sure this is the only way, not like with mt7621 ?

**Comment by rotanid** [comment]
maybe you could link the openwrt wiki article in your PR text above if you mention it.

i would suggest to remove the broken flag and instead mark this PR as broken so it isn't merged before #1626

File: package/gluon-core/luasrc/lib/gluon/upgrade/010-primary-mac (hunk 1 above)
Comment (mweinelt) [reply package/gluon-core/luasrc/lib/gluon/upgrade/010-primary-mac]: And mt7620 before mt7621.

Comment (Dark4MD) [reply package/gluon-core/luasrc/lib/gluon/upgrade/010-primary-mac]: i tried the other ways, phy1/macaddress is off by one in the last digit, only the address on eth0 is matching the one on the case.

**Comment by mweinelt** [comment]
When you're done please squash your changes into a single commit.

**Comment by Dark4MD** [comment]
I open a new, clean PR

**Pull request closed without merging**